차한솔/week2/베스트앨범.js

```diff
@@ -0,0 +1,33 @@
+function solution(genres, plays) {
+	const allGenresWithPlays = new Map();
+	
+	for (let i = 0; i < genres.length; i++) {
+		const genre = genres[i];
+
+		if (!allGenresWithPlays.has(genre)) {
+			allGenresWithPlays.set(genre, {
+				totalPlays: plays[i],
+				songs: [{index: i, plays: plays[i]}],
+			})
+
+			continue;
+		}
+
+		const genreWithPlay = allGenresWithPlays.get(genre)
+
+		allGenresWithPlays.set(genre, {
+			totalPlays: genreWithPlay.totalPlays + plays[i],
+			songs: [...genreWithPlay.songs, {index: i, plays: plays[i]}]
+		})
+	}
+
+	const sortedByTotalPlays = [...allGenresWithPlays].sort((a,b) => b[1].totalPlays - a[1].totalPlays);
```

Comment:
현재 hashMap의 key(genre)는 밑에 로직에서도 사용하지 않으므로,
```suggestion
	const sortedByTotalPlays = [...allGenresWithPlays.values()].sort((a,b) => b.totalPlays - a.totalPlays);
```
이렇게 코드를 바꾸면 조금 더 직관적이고 불필요한 변수를 신경쓰지 않아도 될 꺼 같습니다.